rclpy/rclpy/exceptions.py

```diff
@@ -65,3 +65,29 @@ class InvalidServiceNameException(NameValidationException):
 
     def __init__(self, name, error_msg, invalid_index, *args):
         NameValidationException.__init__(self, 'service name', name, error_msg, invalid_index)
+
+
+# TODO(jubeira): Polish parameter-related exceptions; improve error messages.
+class ParameterNotDeclaredException(Exception):
```

Comment:
Consider adding a `ParameterImmutableException` (analogous to rclcpp) for read only parameters.